app/views/handlers/confirmation_email.py

```diff
@@ -63,16 +65,16 @@ def get_page_title(self):
         return self.page_title
 
     def _publish_fulfilment_request(self):
-        topic_id = current_app.config["EQ_SUBMISSION_CONFIRMATION_TOPIC_ID"]
-        fulfilment_request = ConfirmationEmailFulfilmentRequest(
+        fulfilment_request = ConfirmationEmailTask(
             self.form.email.data, self._session_store.session_data, self._schema
         )
         try:
-            return current_app.eq["publisher"].publish(
-                topic_id, message=fulfilment_request.message
+            return current_app.eq["task-client"].create_task(
+                payload=fulfilment_request.payload,
+                queue_name=EQ_SUBMISSION_CONFIRMATION_QUEUE,
             )
-        except PublicationFailed:
-            raise ConfirmationEmailFulfilmentRequestPublicationFailed
+        except PublicationFailed as exc:
+            raise ConfirmationEmailTaskPublicationFailed from exc
```

Comment:
`create_task` doesn't raise a `PublicationFailed` exception, I think this should be catching `CloudTaskCreationFailed` exceptions.

There is also some inconsistencies in naming between `ConfirmationEmailTaskPublicationFailed` and `CloudTaskCreationFailed` (publication vs creation).